numpy/_core/src/umath/string_ufuncs.cpp

```diff
@@ -363,10 +369,24 @@ string_replace(Buffer<enc> buf1, Buffer<enc> buf2, Buffer<enc> buf3, npy_int64 c
             buf1 += len2;
         }
     }
+    else {  // If match string empty, interleave.
+        while (count > 0) {
+            buf3.buffer_memcpy(out, len3);
+            out += len3;
+            buf1.buffer_memcpy(out, 1);
```

Comment:
I may be missing something and didn't have time to test myself, but I don't think this can be right: you are copying 1 character here, but not updating `out` for that. Since you also do not update `buf1`, you just copy that same character again.

Note that given that `count` is limited to `len(buf1)` for this case, after this copy for the final `count` and after updating `out` and `buf1`, you are guaranteed to be at `end`, so presumably `buf1.buffer_memcpy(out, end1-buf1)` will do nothing.

Reply:
> you are copying 1 character here, but not updating out for that.

We update `buf1` and `out` right after we check that this is not the last iteration.

> Note that given that count is limited to len(buf1)

It's not in the case of `replace('', '', 'A', MAX)`.